### PR TITLE
update most of the dependencies

### PR DESCRIPTION
**Pull request by ordian**
The update was done via `cargo upgrade --workspace` and then a couple of updates were reverted:
* `bitvec` requires update from `parity-scale-codec`
* `shared_memory` update is more involved (seems to be split into https://github.com/elast0ny/raw_sync-rs)
* `tokio` 0.3 was reverted to 0.2, require update from substrate 
- [X] `tiny-keccak` was updated manually

as mentioned in https://github.com/paritytech/substrate/pull/7509#issuecomment-725505818, it would be nice to do a burnin.

### CODE REVIEW

**Comment by ordian** [comment]
Haven't noticed any anomalies in the metrics.

File: node/core/candidate-validation/Cargo.toml
Comment (mxinden) [inline node/core/candidate-validation/Cargo.toml:12]: If I [understand `package` correctly](https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#renaming-dependencies-in-cargotoml), one can use it to rename dependencies. Why is it needed here, given that it does not rename the import variable, but instead just repeats the name?

**Review by mxinden — COMMENTED** [review]
Thanks for doing this @ordian!

For what my review is worth, the changes look good to me, apart from the question below. Given my low involvement in the Polkadot code base and the size of this change, I would very much appreciate more eyes on this pull request.

> Haven't noticed any anomalies in the metrics.

Same here. This has been running on a sentry node (`kusama-sentry-ew4-0`) for more than 5 days. Before merging, I would recommend running it on a validator as well. What do you think @ordian?

**Review by tomaka — COMMENTED** [review]
There's indeed a lot of these `package = "..."`, notably for `futures` and `parity-scale-codec`.

LGTM once these are removed.

File: node/core/candidate-validation/Cargo.toml
Comment (ordian) [reply node/core/candidate-validation/Cargo.toml:12]: This was a bug in `cargo-edit` tool that I used to automatically bump all the dependencies. Reverted this change manually for `futures` in 3d16db4, thanks for catching this for `parity-scale-codec`, will fix now.

**Comment by ordian** [comment]
Addressed the renaming for `parity-scale-codec` and the remaining for `futures`.

Note that this PR doesn't upgrade dependencies constrained by sub-dependencies, like `libp2p` (constrained by `substrate`) or `bitvec` (by `parity-scale-codec`). Once we upgrade dependencies in substrate, there might be a new bunch to upgrade here.

**Comment by ordian** [comment]
> Before merging, I would recommend running it on a validator as well. What do you think @ordian?

I don't think a validator burn-in is necessary here, but don't have a strong opinion either.

**Review by mxinden — APPROVED** [review]
For what my review is worth, this looks good to me.

**Review by tomaka — APPROVED** [review]

**Pull request merged**
